modules/microshift-nw-ipv6-single-stack-config.adoc

```diff
@@ -0,0 +1,93 @@
+// Module included in the following assemblies:
+//
+// * microshift_configuring/microshift-using-config-tools.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-configuring-ipv6-single-stack-config_{context}"]
+= Configuring IPv6 single-stack
+
+You can use the IPv6 network protocol by updating the {microshift-short} service configuration file.
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You have root access to the cluster.
+* Your cluster uses the OVN-Kubernetes network plugin.
+//if you have multus, would you also use the same procedure?
+
+//* The cluster node has IPv6 addresses.
+//also does not need to be pre-config, because we are doing that here, correct?
+//* You have configured an IPv6-enabled router based on your infrastructure.
+//microshift users don't need to pre-config, correct?
+
+.Procedure
+
+. If you have not done so, make a copy of the provided `config.yaml.default` file in the `/etc/microshift/` directory, renaming it `config.yaml`. Keep the new {microshift-short} `config.yaml` you create in the `/etc/microshift/` directory. The new `config.yaml` is read whenever the {microshift-short} service starts. After you create it, the `config.yaml` file takes precedence over built-in settings.
+//Q: would we add two sections for dual stack?
+. Replace the default values in the `network` section of the {microshift-short} YAML with your desired valid values.
++
+.Example single-stack `IPv6` configuration
+[source,yaml]
+----
+apiServer:
+# ....
+network:
+  clusterNetwork:
+  - fd01::/48 <1>
+  serviceNetwork:
+  - fd02::/112 <2>
+node:
+  nodeIP: 2600:1f14:1c48:ee00:2d76:3190:5bc2:5aef <3>
+# ....
+----
+<1> Specify a `clusterNetwork` with a CIDR value that is less than `64`.
+<2> Specify an IPv6 CIDR with a prefix of `112`. Kubernetes uses only the lowest 16 bits. For a prefix of `112`, IP addresses are assigned from `112` to `128` bits.
+<3> Example node IP address. Must be an IPv6 address family.
+
+. To apply networking changes, restart {microshift-short} by entering the following command:
++
+[source,terminal]
+----
+$ sudo systemctl restart microsohift
```

Comment:
This one depends. Changing networks after MicroShift has started at least once is not supported (unless migrating to dual stack). So if MicroShift had started before you need a full cleanup before restarting again. This implies application data loss.
Bottom line, the IP family you choose as the first one can not be changed without a cleanup.